### PR TITLE
Add script to generate bpf2c expected files

### PR DESCRIPTION
**Pull request by saxena-anurag**
## Description

If any change is done to bpf2c output, all the expected files need to be regenerated. This PR adds a script which can be manually run to update all the expected files. The script also trims the absolute file path from the generated code so that next iterations do not show large diffs.

The script can be executed as follows:
`.\scripts\generate_expected_bpf2c_output.ps1 .\x64\Debug\`

## Testing

Current tests will cover it.

## Documentation

NA

### CODE REVIEW

File: tests/bpf2c_tests/expected/bindmonitor_dll.txt
Comment (Alan-Jowett) [inline tests/bpf2c_tests/expected/bindmonitor_dll.txt:1]: Why did this line change?

Comment (saxena-anurag) [reply tests/bpf2c_tests/expected/bindmonitor_dll.txt:1]: No idea. maybe conversion to UTF-8 caused this?

Comment (saxena-anurag) [reply tests/bpf2c_tests/expected/bindmonitor_dll.txt]: Conversion to UTF8 caused extra character to be added at the start. Switching to ASCII fixed it.

**Review by dthaler — COMMENTED** [review]
Please add documentation of what this script is for and how to run it.  (The information is in the PR description but not in any file in the PR.)   It may be as simple as adding it in `#` comments at the top of the script file.   It may also be useful to add a `//` comment referring to the script in bpf_code_generator.cpp

**Comment by Alan-Jowett** [comment]
Thre are bug fixes between LLVM 10 and LLVM 13 where LLVM 10 generates incorrect debug info with wrong line numbers. Please regenerate with LLVM 13.

**Comment by dthaler** [comment]
> Thre are bug fixes between LLVM 10 and LLVM 13 where LLVM 10 generates incorrect debug info with wrong line numbers. Please regenerate with LLVM 13.

@Alan-Jowett https://github.com/microsoft/ebpf-for-windows/blob/main/docs/GettingStarted.md says (under Prerequisites) to use LLVM 10.  Do we need to update that page?

**Comment by Alan-Jowett** [comment]
> > Thre are bug fixes between LLVM 10 and LLVM 13 where LLVM 10 generates incorrect debug info with wrong line numbers. Please regenerate with LLVM 13.
> 
> @Alan-Jowett https://github.com/microsoft/ebpf-for-windows/blob/main/docs/GettingStarted.md says (under Prerequisites) to use LLVM 10. Do we need to update that page?

Yes, we probably should. LLVM 10.0.0 is fairly old now. I don't see any long-term support branch for LLVM, but I think it will be simpler to recommend the version that is pre-installed on the GitHub runners.

**Comment by saxena-anurag** [comment]
> 

One thing though, Cilium XDP program verification fails with LLVM 12 (not tried with LLVM 13). To unblock this PR, I will regenerate using v13, but it looks like Issue #832 will get blocked on verification failures.

**Comment by saxena-anurag** [comment]
> Please add documentation of what this script is for and how to run it. (The information is in the PR description but not in any file in the PR.) It may be as simple as adding it in `#` comments at the top of the script file. It may also be useful to add a `//` comment referring to the script in bpf_code_generator.cpp

Added comment in the script, bpf_code_generator.cpp and in the sample programs.

**Comment by dthaler** [comment]
> > > Thre are bug fixes between LLVM 10 and LLVM 13 where LLVM 10 generates incorrect debug info with wrong line numbers. Please regenerate with LLVM 13.
> > 
> > 
> > @Alan-Jowett https://github.com/microsoft/ebpf-for-windows/blob/main/docs/GettingStarted.md says (under Prerequisites) to use LLVM 10. Do we need to update that page?
> 
> Yes, we probably should. LLVM 10.0.0 is fairly old now. I don't see any long-term support branch for LLVM, but I think it will be simpler to recommend the version that is pre-installed on the GitHub runners. [Windows-2019](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md)

Filed #962

**Review by Alan-Jowett — APPROVED** [review]

File: scripts/generate_expected_bpf2c_output.ps1
Comment (shankarseal) [inline scripts/generate_expected_bpf2c_output.ps1:17]: please add a comment on what this function is filtering for and why.

File: scripts/generate_expected_bpf2c_output.ps1
Comment (shankarseal) [inline scripts/generate_expected_bpf2c_output.ps1:52]: please add generated files for sock_addr and sockops.c

File: scripts/generate_expected_bpf2c_output.ps1
Comment (shankarseal) [inline scripts/generate_expected_bpf2c_output.ps1:1]: general comment on the function naming. I think these verbs are not approved. VSCode uses PSScriptAnalyzer(PSUseApprovedVerbs) and complains about non-approved verbs. Also the style for PS is verb-object such as Install-MyComponent etc.

I dont want to get into battle about the casing. The CICD scripts at least does not use snake casing.

File: scripts/generate_expected_bpf2c_output.ps1
Comment (saxena-anurag) [reply scripts/generate_expected_bpf2c_output.ps1]: Done.

File: scripts/generate_expected_bpf2c_output.ps1
Comment (saxena-anurag) [reply scripts/generate_expected_bpf2c_output.ps1:1]: Changed to Verb-Object

File: scripts/generate_expected_bpf2c_output.ps1
Comment (saxena-anurag) [reply scripts/generate_expected_bpf2c_output.ps1:63]: Added.

**Review by Alan-Jowett — APPROVED** [review]

**Review by shankarseal — APPROVED** [review]

**Pull request merged**
